opencog/scm/CMakeLists.txt

```diff
@@ -33,6 +33,18 @@ ADD_GUILE_MODULE (FILES
 	MODULE_DESTINATION "${GUILE_SITE_DIR}/opencog"
 )
 
+# Create three copies of (opencog as-config) module:
+SET(SCM_IN_BUILD_DIR "#t")
+# One for build dir
+CONFIGURE_FILE(opencog/as-config_TEMPLATE.scm ${GUILE_BIN_DIR}/opencog/as-config.scm)
+# and one for in-place testing of source scm modules
+CONFIGURE_FILE(opencog/as-config_TEMPLATE.scm ${CMAKE_SOURCE_DIR}/opencog/scm/opencog/as-config.scm)
```

Comment:
Ideally we'd remove this, and change all the unit tests to use the build dir.

But this would mean that you couldn't just edit a scm file and immediately rerun a test executable... you'd have to run make to copy your scm files to the build dir first.

I'm not sure if that functionality should be preserved.

Reply:
The cmake ALREADY copies about half of the scm files into the build dir.   The reason it does not copy all of them is because, at some point in history, as people hacked on this, something changed, and the tests started passing without the copy being necessary.    It would be cleanest to either remove the copying (since its currently a big confusing no-op that litters copies of files throughout the build tree) or make it into a requirement.

Reply:
One reason for copying to the build dir, is to make sure all the scm files are laid out in a consistent module hierarchy. This is reflective of what the install dir layout will be, and so this layout is ideally what the tests should use.

An alternative to ensure people can still test scm files without explicitly rerunning make, would be to create a symbolic link from the build dir to the relevent file in the source dir. Then there would be no reason to avoid tests only using the build dir.